### ISSUE TITLE
PSA hierarchical storage: directory listing

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#6366]
Implement directory listing for [PSA hierarchical storage](https://github.com/Mbed-TLS/mbedtls/issues/6363).

Deliverables: implementation of these functions, and unit tests. Using the new API outside of its own unit tests is out of scope.

Prerequisite: https://github.com/Mbed-TLS/mbedtls/issues/6364
